torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2331,7 +2308,8 @@ def state_dict(self, *args, **kwargs):
         if torch.cuda.is_available():
             torch.cuda.synchronize()
         self._lazy_init()
-        self._clear_grads_if_needed()
```

Comment:
Before this was a no-op if not root. Adding `if self._is_root` does not change semantics.